### ISSUE TITLE
Export bundle error because of missing files

### ISSUE THREAD

**Issue by sporkd** [MeasureAuthoringTool/bonnie#240]
Hi. When I attempt to export a bundle I'm getting the following error:

```
  10:53:18 web.1  | TypeError (no implicit conversion of nil into String):
  10:53:18 web.1  |   app/controllers/admin/users_controller.rb:56:in `bundle'
```

It appears the reason is the bonnie_bundler gem is trying to read some files that don't exist in `/db/measures/html` and `/db/measures/hqmf`. [Here's where it's breaking](https://github.com/projecttacoma/bonnie_bundler/blob/master/lib/measures/exporter/bundle_exporter.rb#L167-L168)

Is there somewhere I'm supposed to get these files, or am I missing a step before trying to export?

Thanks in advance!

**Comment by andrequina** [comment]
Can you get a full stack trace?

Those two files should have a rescue so I don't think they would be the source of the exception.

rescue BonnieBundler.logger.warn("\tNo source HTML for #{measure.measure_id}")

The source HTML and HQMF are nice to have but they are not required for the bundle.  When a Cypress bundle is built we store these files in the bundle so that we have them in case we need to debug an issue in the HQMF Parser etc.  However, they are not required downstream in popHealth or Cypress.

**Comment by andrequina** [comment]
It may be coming out of the following lines:

```
      export_file File.join(path, "#{measure.measure_id}.html"),html
      export_file File.join(path, "hqmf1.xml"), hqmf1
```

these should probably have trailing checks for the existence of the html and hqmf files.

**Comment by sporkd** [comment]
Oh you're right about the rescues @andrequina. I guess it is getting past that. But hard to tell from the stack trace exactly where. But I don't ever see `Exporting patient ...` getting logged:

```
  11:44:22 web.1  | Started GET "/admin/users/53478b639c65701d8e000001/bundle" for 127.0.0.1 at 2014-04-11 11:44:21 -0700
  11:44:22 web.1  | Processing by Admin::UsersController#bundle as HTML
  11:44:22 web.1  |   Parameters: {"id"=>"53478b639c65701d8e000001"}
  11:44:22 web.1  |   MOPED: 127.0.0.1:17017 QUERY        database=bonnie_development collection=users selector={"$query"=>{"_id"=>"53478b639c65701d8e000001"}, "$orderby"=>{:_id=>1}} flags=[:slave_ok] limit=-1 skip=0 batch_size=nil fields=nil (0.6568ms)
  11:44:22 web.1  |   MOPED: 127.0.0.1:17017 QUERY        database=bonnie_development collection=users selector={"_id"=>"53478b639c65701d8e000001"} flags=[:slave_ok] limit=0 skip=0 batch_size=nil fields=nil (0.4017ms)
  11:44:22 web.1  | Exporting measures
  11:44:22 web.1  |   MOPED: 127.0.0.1:17017 QUERY        database=bonnie_development collection=draft_measures selector={"user_id"=>"53478b639c65701d8e000001"} flags=[:slave_ok] limit=0 skip=0 batch_size=nil fields={"hqmf_id"=>1} (0.4730ms)
  11:44:22 web.1  |   MOPED: 127.0.0.1:17017 QUERY        database=bonnie_development collection=measures selector={"hqmf_id"=>{"$in"=>["8A4D92B2-3946-CDAE-0139-7941863D7FFD"]}} flags=[:slave_ok] limit=0 skip=0 batch_size=nil fields=nil (0.5529ms)
  11:44:22 web.1  | Exporting sources
  11:44:22 web.1  |   MOPED: 127.0.0.1:17017 QUERY        database=bonnie_development collection=draft_measures selector={"user_id"=>"53478b639c65701d8e000001"} flags=[:slave_ok] limit=0 skip=0 batch_size=nil fields=nil (3.3240ms)
  11:44:22 web.1  |   No source HTML for 
  11:44:22 web.1  |   No source HQMFv1 for 
  11:44:22 web.1  | Completed 500 Internal Server Error in 1274ms
  11:44:22 web.1  | 
  11:44:22 web.1  | TypeError (no implicit conversion of nil into String):
  11:44:22 web.1  |   app/controllers/admin/users_controller.rb:56:in `bundle'
  11:44:22 web.1  | 
  11:44:22 web.1  | 
  11:44:22 web.1  |   Rendered vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.14/lib/action_dispatch/middleware/templates/rescues/_trace.erb (2.6ms)
  11:44:22 web.1  |   Rendered vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.14/lib/action_dispatch/middleware/templates/rescues/_request_and_response.erb (1.4ms)
  11:44:22 web.1  |   Rendered vendor/bundle/ruby/2.0.0/gems/actionpack-3.2.14/lib/action_dispatch/middleware/templates/rescues/diagnostics.erb within rescues/layout (14.0ms)
```

**Comment by andrequina** [comment]
@sporkd I can reproduce.  I will look into this.

**Comment by andrequina** [comment]
@sporkd there were a bunch of issues with the bundle export.  I guess it was in rougher shape than I thought.  The following pull request should fix them.

https://github.com/projecttacoma/bonnie_bundler/pull/25

We should be able to get this pull request merged in early next week.

I have not tested the exported bundle with popHealth yet, so if that's what you are working towards there may be issues bringing the bundle into popHealth.  I think it should work, but I have not tested it yet.

**Comment by sporkd** [comment]
Awesome @andrequina! Thanks for the quick fix.  I will test it out with popHealth and report back.

**Comment by sporkd** [comment]
@andrequina Update. After applying your changes, I was able to successfully export a bundle from bonnie and then import it into popHealth using the bundle:import rake task.  The only hangup I ran into is the exported measure json from bonnie has category set to null.  Category is expected by the popHealth UI to select/deselect measures.  So I had to manually set that field in the rails console before I could see any measures.  Otherwise looks good. Thanks again!
